File: blogPessoal/src/main/java/org/generation/blogPessoal/model/Postagem.java
package org.generation.blogPessoal.model;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

@Data
@Entity
@Table(name = "postagens")
public class Postagem {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	
	@NotBlank
	@Size(min = 5, max = 100)
	private String titulo;

	@NotBlank
	private String texto;

	@Temporal(TemporalType.TIMESTAMP)
	private Date data = new java.sql.Date(System.currentTimeMillis());
	
	@ManyToOne
	@JsonIgnoreProperties("postagem")
	private Tema tema;

}

File: Projeto Pessoal Front-End/blogPessoal/src/app/model/Postagem.ts
import { Tema } from "./Tema"
import { Usuario } from "./Usuario"

export class Postagem {
    public id: number
    public imagem: string
    public titulo: string
    public texto: string
    public data: Date
    public usuario: Usuario
    public curtida: number
    public amou: number
    public tema: Tema
    public listaCurtidaUsuario: Usuario[]
    public listaAmouUsuario: Usuario[]
}